### ISSUE TITLE
Subdirectory field not working / causes exception

### ISSUE THREAD

**Issue by alertedsnake** [awslabs/aws-codedeploy-plugin#28]
No matter what I put into the "subdirectory" field, I get this exception - even when I leave it blank.

Suggestions?

```
Failed CodeDeploy post-build step; exception follows.
Provided path is not a subdirectory of the workspace: /var/lib/jenkins-slave/workspace/TestApp-CodeDeploy-Test
java.lang.IllegalArgumentException: Provided path is not a subdirectory of the workspace: /var/lib/jenkins-slave/workspace/TestApp-CodeDeploy-Test
	at com.amazonaws.codedeploy.AWSCodeDeployPublisher.getSourceDirectory(AWSCodeDeployPublisher.java:242)
	at com.amazonaws.codedeploy.AWSCodeDeployPublisher.perform(AWSCodeDeployPublisher.java:215)
	at hudson.tasks.BuildStepMonitor$2.perform(BuildStepMonitor.java:32)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:726)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:671)
	at hudson.model.Run.execute(Run.java:1766)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
Build step 'Deploy an application to AWS CodeDeploy' marked build as failure
Finished: FAILURE
```

**Comment by grusy** [comment]
I have the same issue when the job is executed on a slave. Working well on the master node.

**Comment by alertedsnake** [comment]
Yeah I should have mentioned this is on a slave.

**Comment by JohnBloom** [comment]
I have the same problem. Anybody have a workaround? I am trying to get this working on a windows slave.

**Comment by JohnBloom** [comment]
Also I believe the exception is backwards: 

`throw new IllegalArgumentException("Provided path is not a subdirectory of the workspace: " + sourcePath );`

It should be:

`throw new IllegalArgumentException("Provided path is not a subdirectory of the workspace: " + basePath );`

But to really diagnose the problem it would be nice if we could see both paths:

`throw new IllegalArgumentException("Provided path: " + sourcePath + " is not a subdirectory of the workspace: " + basePath );`

**Comment by ramakantjoshi** [comment]
was this resolved? I see the same error

**Comment by JohnBloom** [comment]
It has not been resolved. I am not sure if this project is being actively developed.

**Comment by defond0** [comment]
I am having the same issue, would love to see how this shakes out, ++on wanting to see both halves of the path in the exception to properly diagnose

**Comment by djcrabhat** [comment]
:+1:, would love to figure this out

**Comment by JohnBloom** [comment]
I found this fix on a fork. Perhaps @ydubreuil would be willing to provide a pull request. https://github.com/ydubreuil/aws-codedeploy-plugin/commit/84958cc2bc37905ebcaf1efa69cc0b757bb2cc4c

**Comment by ydubreuil** [comment]
I sent a PR two month ago https://github.com/jenkinsci/aws-codedeploy-plugin/pull/3

**Comment by suryanarayanan** [comment]
I've merged the fix in both jenkinsci and awslabs repos.

**Comment by JohnBloom** [comment]
does it automatically push out a new version to the jenkins package store or does that still need to be done?

**Comment by suryanarayanan** [comment]
I've released the new version of the plugin 1.7 to the jenkins plugin store. It should be available in a while. It should contain the subdirectory fix, logging tweaks and support for our latest region ap-northeast-1.

**Comment by JohnBloom** [comment]
I can confirm that this is working on the new version. Thanks for all the hard work @suryanarayanan and @ydubreuil!
